File: src/main/java/pe/nanamochi/banchus/packets/server/MatchPlayerFailedPacket.java
package pe.nanamochi.banchus.packets.server;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.packets.Packet;
import pe.nanamochi.banchus.packets.Packets;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MatchPlayerFailedPacket implements Packet {
  private int slotId;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_MATCH_PLAYER_FAILED;
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/MatchPlayerFailedPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.BanchoPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataWriter

class MatchPlayerFailedPacket(var slotId: Int = 0) : BanchoPacket(), BanchoPacket.Server {
    override val type = PacketType.BANCHO_MATCH_PLAYER_FAILED

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeInt32(out, slotId)
    }
}
